src/tools/rbd_mirror/image_replayer/BootstrapRequest.cc

```diff
@@ -194,6 +194,10 @@ void BootstrapRequest<I>::handle_get_client(int r) {
     derr << ": failed to retrieve client: " << cpp_strerror(r) << dendl;
     finish(r);
     return;
+  } else if (m_client.state == cls::journal::CLIENT_STATE_DISCONNECTED) {
+    dout(0) << ": client flagged disconnected, re-registering" << dendl;
+    unregister_client();
```

Comment:
I think you can just fail with some special result code that ```ImageReplayer``` can use to know that it was disconnected and should stop (with error).